### PR TITLE
test(python): refactored `write_database` unit tests to properly separate concerns

### PR DESCRIPTION
**Pull request by alexander-beedie**
Ref: [#10763](https://github.com/pola-rs/polars/pull/10763#issuecomment-1696945452).

Splits the single "write database" unit test into the following:

* `test_write_database_create`
* `test_write_database_append`
* `test_write_database_create_quoted_tablename`
* `test_write_database_errors`

Due to https://github.com/apache/arrow-adbc/issues/1000, the `test_write_database_create_quoted_tablename` check is skipped when run for `adbc_driver_sqlite` version 0.6.0.

### CODE REVIEW

File: py-polars/tests/unit/io/test_database_write.py [hunk 1]
```diff
@@ -11,93 +12,124 @@
...
+    reason="adbc_driver_sqlite not available below Python 3.9 / on Windows",
+)
+def test_write_database_create_quoted_tablename(
+    engine: DbWriteEngine, tmp_path: Path
+) -> None:
+    if engine == "adbc" and adbc_driver_sqlite.__version__ == "0.6.0":  # type: ignore[attr-defined]
+        # see: https://github.com/apache/arrow-adbc/issues/1000
+        pytest.skip(
+            reason="ADBC SQLite driver v0.6.0 has a bug with quoted/qualified table names"
+        )
```
Comment (stinodego) [inline py-polars/tests/unit/io/test_database_write.py:104]: This can also just be a `@pytest.mark.skipif`, right?

**Review by stinodego — COMMENTED** [review]
Thanks for this! I just have one minor comment, otherwise good to go.

File: py-polars/tests/unit/io/test_database_write.py (hunk 1 above)
Comment (alexander-beedie) [reply py-polars/tests/unit/io/test_database_write.py:104]: Ahh, can you can reference the value of a parameterised param in the skipIf? If so, then "yes", but I didn't realise you could do that (we still want the test to run for sqlalchemy).

Comment (stinodego) [reply py-polars/tests/unit/io/test_database_write.py:104]: Oh, maybe you can't... wanna find out? :smile:

Comment (alexander-beedie) [reply py-polars/tests/unit/io/test_database_write.py:104]: > Oh, maybe you can't... wanna find out? 😄

Lol... well, when I (very briefly!) checked, the StackOverflow suggestion was an inline `pytest.skip` 😅

**Review by stinodego — APPROVED** [review]
Thanks!

**Pull request merged**